cocotb/share/makefiles/Makefile.inc

```diff
@@ -76,6 +76,14 @@ BASE_WARNS := -Wall -Wextra -Wconversion -Wcast-qual -Wwrite-strings
 CC_WARNS   := $(BASE_WARNS) -Wstrict-prototypes -Waggregate-return
 CXX_WARNS  := $(BASE_WARNS) -Wnon-virtual-dtor -Woverloaded-virtual
 
+# version-specific flags
+# easiest to compare versions with python
+version_less_than_equal = $(shell $(PYTHON_BIN) -c "import sys; from distutils.version import LooseVersion; print(LooseVersion(sys.argv[1]) <= LooseVersion(sys.argv[2]))" $(1) $(2))
+GXX_VERSION = $(shell g++ -dumpversion)
+ifeq ($(call version_less_than_equal,4.9,$(GXX_VERSION)),True)
+CXX_WARNS += -Wsuggest-override
```

Comment:
That won't fly, unfortunately. People can use clang and other compilers. I see two ways out:
- Remove this check, and only test in CI by modifying CXX_ARGS (preferred).
- Test for the feature instead of checking the version.

Reply:
I was under the impression that we had g++ hard-coded anyway, but perhaps we don't want to entrench ourselves further there.

Reply:
We had, but the recent work by @ktbarrett removed that explicit dependency.